cocos/platform/CMakeLists.txt

```diff
@@ -22,7 +22,16 @@
 # THE SOFTWARE.
 # ****************************************************************************/
 
-if(WINDOWS)
+if(ANDROID)
```

Comment:
In this file should be only:
```
add_subdirectory(android)
add_subdirectory(apple)
```
sources should be added inside android/apple

Reply:
the reasone to add `cocos/platform/android/CMakeLists.txt` file is we need to `whole-archive` part sources, or symble might be striped when linking.

some developer make feedbacks that those games works well on android when build by `ndk-build`, but crash sometimes when build by `ndk-build`. espicailly execute some `lua` method related with `CCLuaJavaBridge.cpp` on __cocos2d-x3.17__. By compare with `ndk-build`, I found  `whole-archive` is the point.

Reply:
Please add comment with link to issues explaining why whole-archive is needed

Reply:
https://github.com/cocos2d/cocos2d-x/blob/5c381d8ee60078b1ff8380b794836ea61d4ed2ee/cocos/Android.mk#L281

https://github.com/cocos2d/cocos2d-x/blob/5c381d8ee60078b1ff8380b794836ea61d4ed2ee/cocos/platform/android/Android.mk#L5

see above, ndk-build use the `LOCAL_WHOLE_STATIC_LIBRARIES ` for years, use `whole-archive` in cmake to aim same logic.

> no repo issues about it, I think we should keep the logic no change